### PR TITLE
Added Recipient Lists resource lib

### LINKED ISSUE #12: Create wrapper for recipient list API
See https://www.sparkpost.com/api#/reference/recipient-lists for docs for all resources.

### PR DESCRIPTION
**Pull request by aydrian**
Closes #12

Take a look and see if we want to abstract it any further. I left passing the recipients objects as is.

### CODE REVIEW

**Comment by richleland** [comment]
Running `make docs` throws some errors:

```bash
make docs-open                                                                                                        2015-08-28 09:29
. venv/bin/activate; cd docs && make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.3.1
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 0 source files that are out of date
updating environment: 2 added, 3 changed, 2 removed
reading sources... [100%] resources/recipient_lists
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/api/recipient_lists.rst:4: WARNING: Title underline too short.

:mod:`sparkpost.recipient_lists`
=============================
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/api/transmissions.rst:6: WARNING: autodoc: failed to import class u'Transmission' from module u'sparkpost.transmission'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/venv/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
ImportError: No module named transmission
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/resources/recipient_lists.rst:2: WARNING: Title underline too short.

Recipient Lists
=============
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/resources/recipient_lists.rst:41: WARNING: Title underline too short.

Retrieve a recipient list
-----------------------
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/resources/recipient_lists.rst:53: WARNING: Title underline too short.

List all recipient lists
----------------------
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/resources/recipient_lists.rst:53: WARNING: Title underline too short.

List all recipient lists
----------------------
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/resources/recipient_lists.rst:81: ERROR: Unknown target name: "sparkpost recipient list api reference".
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] resources/recipient_lists
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/sparkpost/recipient_lists.py:docstring of sparkpost.recipient_lists.RecipientLists.create:None: WARNING: more than one target found for cross-reference u'list': sparkpost.recipient_lists.RecipientLists.list, sparkpost.templates.Templates.list
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/sparkpost/recipient_lists.py:docstring of sparkpost.recipient_lists.RecipientLists.update:None: WARNING: more than one target found for cross-reference u'list': sparkpost.recipient_lists.RecipientLists.list, sparkpost.templates.Templates.list
generating indices... genindex py-modindex
writing additional pages... search
copying static files... WARNING: html_static_path entry u'/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 10 warnings.

Build finished. The HTML pages are in _build/html.
. venv/bin/activate; open docs/_build/html/index.html
```

File: docs/resources/recipient_lists.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+See the `python-sparkpost recipient_lists examples`_.
+
+.. _python-sparkpost recipient_lists examples: https://github.com/SparkPost/python-sparkpost/tree/master/examples/recipient_lists
+
+
+Additional documentation
+------------------------
+
+See the `SparkPost Recipient List API Reference`_.
```
Comment (richleland) [inline docs/resources/recipient_lists.rst]: This link is broken in the HTML output by `make docs`.

Comment (richleland) [inline docs/resources/recipient_lists.rst]: It's `Recipient List` vs. `Recipient Lists`

File: sparkpost/recipient_lists.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        results = self.request('PUT', uri, data=json.dumps(payload))
+        return results
+
+    def delete(self, list_id):
+        """
+        Delete a recipient list by ID
+
+        :param str list_id: ID of the recipient list you want to delete
+
+        :returns: TODO
```
Comment (richleland) [inline sparkpost/recipient_lists.py]: Returns True if successful.

File: sparkpost/recipient_lists.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import json
+
+from .base import Resource
+
+
+class RecipientLists(Resource):
+    """
+    RecipientLists class used to create, update, delete, list and get recipient
+    lists. For
```
Comment (richleland) [inline sparkpost/recipient_lists.py]: What's with the weird line wrap?

File: sparkpost/recipient_lists.py (hunk 2 above)
Comment (richleland) [inline sparkpost/recipient_lists.py]: Ugh nevermind - this is one of the APIs that returns 200 with `{}` - we should just put empty `dict`.

**Comment by richleland** [comment]
Looks like just docs issues - other than that good to go.

**Comment by richleland** [comment]
I see the following errors:

```bash
make docs-open                                                                                                        2015-08-28 15:24
. venv/bin/activate; cd docs && make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.3.1
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 2 source files that are out of date
updating environment: 0 added, 3 changed, 0 removed
reading sources... [100%] resources/recipient_lists
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/api/transmissions.rst:6: WARNING: autodoc: failed to import class u'Transmission' from module u'sparkpost.transmission'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/venv/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 385, in import_object
    __import__(self.modname)
ImportError: No module named transmission
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] resources/recipient_lists
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/sparkpost/recipient_lists.py:docstring of sparkpost.recipient_lists.RecipientLists.create:None: WARNING: more than one target found for cross-reference u'list': sparkpost.recipient_lists.RecipientLists.list, sparkpost.templates.Templates.list
/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/sparkpost/recipient_lists.py:docstring of sparkpost.recipient_lists.RecipientLists.update:None: WARNING: more than one target found for cross-reference u'list': sparkpost.recipient_lists.RecipientLists.list, sparkpost.templates.Templates.list
generating indices... genindex py-modindex
writing additional pages... search
copying static files... WARNING: html_static_path entry u'/Users/rleland/src/oss/reviews/python-sparkpost-aydrian/docs/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 4 warnings.

Build finished. The HTML pages are in _build/html.
. venv/bin/activate; open docs/_build/html/index.html
```

Looks like some references to the old `transmission` class exist.

**Comment by aydrian** [comment]
All doc warnings are resolved with the exception of the `list` method issues. Not sure how to resolve.

**Pull request merged**